refactor(store): remove ActionReducer from 'on' generics and use labe…

markostanimirovic opened the pull request:
…led tuple for arguments

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

timdeschryver reviewed (approved):
Huh, this is weird... I thought that we fixed this when we refactored this to variadic tuple types (based on your review).
Thanks for catching this (again 😅)!

markostanimirovic commented:
> Huh, this is weird... I thought that we fixed this when we refactored this to variadic tuple types (based on your review).
> Thanks for catching this (again 😅)!

Haha yes, it magically disappeared 😁
You're welcome 🙂
